engine/univMinim.ml

```diff
@@ -271,17 +271,23 @@ let is_bound l lbound = match lbound with
 | UGraph.Bound.Prop -> Level.is_prop l
 | UGraph.Bound.Set -> Level.is_set l
 
+(* if [is_minimal u] then constraints [u <= v] may be dropped and get
+   used only for set_minimization. *)
+let is_minimal ~lbound u =
+  Level.is_sprop u || Level.is_prop u || is_bound u lbound
+
 (* TODO check is_small/sprop *)
 let normalize_context_set ~lbound g ctx us algs weak =
   let (ctx, csts) = ContextSet.levels ctx, ContextSet.constraints ctx in
   (* Keep the Prop/Set <= i constraints separate for minimization *)
   let smallles, csts =
-    Constraint.partition (fun (l,d,r) -> d == Le && (is_bound l lbound || Level.is_sprop l)) csts
+    Constraint.partition (fun (l,d,r) -> d == Le && is_minimal ~lbound l) csts
   in
   let smallles = if get_set_minimization ()
     then Constraint.filter (fun (l,d,r) -> LMap.mem r us && not (Level.is_sprop l)) smallles
```

Comment:
At some point for some reason I started to do
```suggestion
    then Constraint.filter (fun (l,d,r) -> LMap.mem r us && not (Level.minimal ~lbound l)) smallles
```
This was broken and necessitated commits like 78c4f3fd292f803e9bc6469260f73e733573931e
Now that I removed it the stdlib passes without change again.